gef.py

```diff
@@ -538,6 +540,9 @@ def __init__(self, elf="", minimalist=False):
         with open(elf, "rb") as fd:
             # off 0x0
             self.e_magic, self.e_class, self.e_endianness, self.e_eiversion = struct.unpack(">IBBB", fd.read(7))
+            if struct.pack(">I", self.e_magic) != Elf.e_magic:
```

Comment:
Note that I added a magic signature check here, which repacked the already unpacked `self.e_magic`.  For better performance, perhaps consider of storing the signature directly as binary.